From 6b781688e93f0dbeb5a0c49688482f4de8e67006 Mon Sep 17 00:00:00 2001
From: Jindrich Novy <jnovy@redhat.com>
Date: Mon, 29 Apr 2013 14:31:27 +0200
Subject: [PATCH 1/3] Use direct path when calling scl_enabled

---
 scl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scl.c b/scl.c
index d02e82f0825ca804c82325111909d445c4758ca3..f64e3ac3be1c70df8f633506a6b74915a1f782b3 100644
--- a/scl.c
+++ b/scl.c
@@ -323,7 +323,7 @@ int main(int argc, char **argv) {
 		size_t r;
 		char scl_dir[BUFSIZ];
 
-		check_asprintf(&enabled, "scl_enabled %s\nif [ $? != 0 ]; then\n"
+		check_asprintf(&enabled, "/usr/bin/scl_enabled %s\nif [ $? != 0 ]; then\n"
 					 "  SCLS+=(%s)\n"
 					 "  export X_SCLS=$(printf '%%q ' \"${SCLS[@]}\")\nfi\n", argv[i], argv[i]);
 		write_script(tfd, enabled);
-- 
1.7.11.7

