diff -r 5da2feeb4b1c -r 03cb1c0a1c57 sphinx/environment.py
--- a/sphinx/environment.py	Mon Oct 29 13:07:21 2012 +0900
+++ b/sphinx/environment.py	Mon Oct 29 21:00:34 2012 +0900
@@ -787,7 +787,12 @@
             app.emit('doctree-read', doctree)
 
         # store time of build, for outdated files detection
-        self.all_docs[docname] = time.time()
+        # note::
+        #  Some filesystem's have coarse timestamp resolution.
+        #  Therefore time.time() is older than filesystem's timestamp.
+        #  ex. FAT32 have 2sec resolution.
+        self.all_docs[docname] = max(
+                time.time(), path.getmtime(self.doc2path(docname)))
 
         if self.versioning_condition:
             # get old doctree
