diff -r a1f50d969fe8 -r 7685daff91f2 sphinx/ext/intersphinx.py
--- a/sphinx/ext/intersphinx.py	Tue Oct 30 10:04:18 2012 +0900
+++ b/sphinx/ext/intersphinx.py	Tue Oct 30 12:44:53 2012 +0900
@@ -191,9 +191,10 @@
         for name, _, invdata in cache.itervalues():
             if name:
                 env.intersphinx_named_inventory[name] = invdata
-            for type, objects in invdata.iteritems():
-                env.intersphinx_inventory.setdefault(
-                    type, {}).update(objects)
+            else:
+                for type, objects in invdata.iteritems():
+                    env.intersphinx_inventory.setdefault(
+                        type, {}).update(objects)
 
 
 def missing_reference(app, env, node, contnode):
